server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationRelocationIT.java

```diff
@@ -419,18 +419,7 @@ public void testNewlyAddedReplicaIsUpdated() throws Exception {
         assertAcked(
             client().admin().indices().prepareUpdateSettings(INDEX_NAME).setSettings(Settings.builder().put(SETTING_NUMBER_OF_REPLICAS, 1))
         );
-
-        ClusterHealthResponse clusterHealthResponse = client().admin()
-            .cluster()
-            .prepareHealth()
-            .setWaitForEvents(Priority.LANGUID)
-            .setWaitForNodes("2")
-            .setWaitForGreenStatus()
-            .setTimeout(TimeValue.timeValueSeconds(2))
-            .execute()
-            .actionGet();
-        assertFalse(clusterHealthResponse.isTimedOut());
-        ensureGreen(INDEX_NAME);
+        ensureYellowAndNoInitializingShards(INDEX_NAME);
```

Comment:
why not use ensureGreen() here? Eventually after adding replica shard the cluster will be green right?

Reply:
Yes you are right, both of these methods would work in this case.